### ISSUE TITLE
macro ilogb on windows

### ISSUE THREAD

**Issue by mglisse** [CGAL/cgal#3440]
`CGAL/config.h`:
```
#if defined(BOOST_MSVC)
#define CGAL_CFG_NO_CPP0X_ISFINITE 1 // used in <CGAL/CORE/Filter.h>
#endif
```
`CGAL/CORE/Filter.h`:
```
#if !defined CGAL_CFG_NO_CPP0X_ISFINITE
  #define CGAL_CORE_finite(x)   std::isfinite(x)
#elif defined (_MSC_VER) || defined (__MINGW32__) // add support for MinGW
  #define CGAL_CORE_finite(x)   _finite(x)
  #define ilogb(x)      (int)_logb(x)
#else
  #define CGAL_CORE_finite(x)   finite(x)
#endif
```
Defining ilogb as a macro is a very bad idea, it breaks code like Boost.Multiprecision, and depending on the inclusion order it might even break standard headers.

**Comment by sloriot** [comment]
I'll push the patch in your branch once the tests are completed on my machine.

**Comment by afabri** [comment]
In fact it made it into the standard with C++11
https://en.cppreference.com/w/cpp/numeric/math/ilogb

**Comment by mglisse** [comment]
@sloriot I think it would also be good to update the definition of CGAL_CFG_NO_CPP0X_ISFINITE for new versions of VS, which do support isfinite (and ilogb). MS doesn't document isfinite anywhere so it is hard to say when it was introduced, but for VC2017 I see `using _CSTD isfinite;` in cmath.

**Comment by sloriot** [comment]
I agree but I wanted to make the most less dangerous change and be as conservative as possible because we did not have any valid result for 3 cycles now.

**Comment by mglisse** [comment]
You can skip my branch for a few cycles if that helps. One of the most interesting platforms (no GMP and no LEDA) is not in the testsuite.

**Comment by sloriot** [comment]
The issue was fixed in #3406 merged a while ago.
